agent/http_test.go

```diff
@@ -724,6 +726,104 @@ func TestParseWait(t *testing.T) {
 		t.Fatalf("Bad: %v", b)
 	}
 }
+func TestPProfHandlers_EnableDebug(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	a := NewTestAgent(t.Name(), "enable_debug = true")
+	defer a.Shutdown()
+
+	resp := httptest.NewRecorder()
+	req, _ := http.NewRequest("GET", "/debug/pprof/profile", nil)
+
+	a.srv.Handler.ServeHTTP(resp, req)
+
+	require.Equal(http.StatusOK, resp.Code)
+}
+func TestPProfHandlers_DisableDebugNoACLs(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	a := NewTestAgent(t.Name(), "enable_debug = false")
+	defer a.Shutdown()
+
+	resp := httptest.NewRecorder()
+	req, _ := http.NewRequest("GET", "/debug/pprof/profile", nil)
+
+	a.srv.Handler.ServeHTTP(resp, req)
+
+	require.Equal(http.StatusUnauthorized, resp.Code)
+}
+
+func TestPProfHandlers_ACLs(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+	dc1 := "dc1"
+
+	a := NewTestAgent(t.Name(), `
+	acl_datacenter = "`+dc1+`"
+	acl_default_policy = "deny"
+	acl_master_token = "master"
+	acl_agent_token = "agent"
+	acl_agent_master_token = "towel"
+	acl_enforce_version_8 = true
+	enable_debug = false
+`)
+
+	cases := []struct {
+		code        int
+		token       string
+		endpoint    string
+		nilResponse bool
+	}{
+		{
+			code:        http.StatusOK,
+			token:       "master",
+			endpoint:    "/debug/pprof/heap",
+			nilResponse: false,
+		},
+		{
+			code:        http.StatusForbidden,
+			token:       "agent",
+			endpoint:    "/debug/pprof/heap",
+			nilResponse: true,
+		},
+		{
+			code:        http.StatusForbidden,
+			token:       "agent",
+			endpoint:    "/debug/pprof/",
+			nilResponse: true,
+		},
+		{
+			code:        http.StatusForbidden,
+			token:       "",
+			endpoint:    "/debug/pprof/",
+			nilResponse: true,
+		},
+		{
+			code:        http.StatusOK,
+			token:       "master",
+			endpoint:    "/debug/pprof/heap",
+			nilResponse: false,
+		},
+		{
+			code:        http.StatusForbidden,
+			token:       "towel",
+			endpoint:    "/debug/pprof/heap",
+			nilResponse: true,
+		},
+	}
+
+	defer a.Shutdown()
+	testrpc.WaitForLeader(t, a.RPC, "dc1")
+
+	for i, c := range cases {
+		req, _ := http.NewRequest("GET", fmt.Sprintf("%s?token=%s", c.endpoint, c.token), nil)
```

Comment:
It would be cool to run each of these in `t.Run()` - it's the normal pattern for table tests and has a few benefits like being able to filter to only certain cases in `-run`.